src/routes/__test__/BaseLayout.test.tsx

```diff
@@ -0,0 +1,106 @@
+import { describe, expect, it, vi } from 'vitest'
+import { render, screen } from '@testing-library/react'
+import { MemoryRouter, Routes, Route } from 'react-router-dom'
+import CssBaseline from '@mui/material/CssBaseline'
+import CustomThemeProvider from '@/theme/CustomThemeProvider'
+import BaseLayout from '../BaseLayout'
+import '@/i18n/config'
+
+// Mock the child components
+vi.mock('@/components/Header/Header', () => ({
+    default: () => <div data-testid="header">Header Component</div>,
+}))
+
+vi.mock('@/components/Footer/Footer', () => ({
+    default: () => <div data-testid="footer">Footer Component</div>,
+}))
+
+const mockFetchData = vi.fn<() => unknown>()
+
+// Mock a child component for testing Outlet
+const MockChildComponent = () => {
+    const data = mockFetchData()
+    if (data instanceof Promise) {
+        // eslint-disable-next-line @typescript-eslint/only-throw-error
+        throw data
+    }
+    return <div data-testid="child-content">Child Content</div>
+}
+
+// Wrapper component for testing BaseLayout with routing
+const TestWrapper = ({ children }: { children: React.ReactNode }) => (
```

Comment:
Have you considered using the `customRender` from `src/tests/customRender.tsx`? 

A lot of it overlaps with what you wrote here